src/optimize/bundles_route/bundles_route.js

```diff
@@ -108,7 +110,13 @@ export function createBundlesRoute({
   ];
 }
 
-function buildRouteForBundles(publicPath, routePath, bundlesPath, fileHashCache) {
+function buildRouteForBundles(
+  publicPath,
+  routePath,
+  bundlesPath,
+  fileHashCache,
+  replacePublicPath = true
```

Comment:
I went down this route originally. Unfortunately, the way `inert` interfaces with hapi is less than ideal.

Basically if we want inert to support looking up `br` encoded files for brotli, we have register an encoder with Hapi (even though it should just be looking up these files from the file system) which causes Hapi to start using the brotli encoder for all responses. This isn't what we want because brotli compression is slow (though decompression is fast).

The reason for this is that [inert calls request.info.acceptEncoding](https://github.com/kanongil/inert/blob/v5.1.0/lib/file.js#L151) which in turn calls this [internal Hapi function](https://github.com/hapijs/hapi/blob/v17.5.3/lib/compression.js#L72) which will only return an encoding if it's registered with Hapi. So even if we add `br` to the `lookupMap` option provided by inert, we still have to add the encoder to the Hapi server globally, which is not what we want.

The only way I was able to get around this was by implementing the file lookup logic manually, inside the dynamicAssetResponse function.

Reply:
Hmm, alright. (honestly I just hate errant boolean arguments because `buildRouteForBundles(..., true)` means nothing to readers

Reply:
I'm also reasonably concerned with the the file handle management logic and would prefer we weren't generating our own etags and such... What if we use the `onPreHandler` to mutate the `path` param to include the `.br` version of the file when it's found and supported by the accept encoding header?

Reply:
> Hmm, alright. (honestly I just hate errant boolean arguments because `buildRouteForBundles(..., true)` means nothing to readers)

I can definitely make these named parameters 👍 

> I'm also reasonably concerned with the the file handle management logic and would prefer we weren't generating our own etags and such... What if we use the `onPreHandler` to mutate the `path` param to include the `.br` version of the file when it's found and supported by the accept encoding header?

I agree. We're already exploring ditching the etag/must-revalidate cache strategy in favor of using max-age + versioned URLs. I'm thinking that if we can remove our need for etag for KP bundles, we can greatly simplify the file handling logic we have since we won't need to read the file twice.

Modifying the `path` param may work as well, but I will experiment with those options outside this PR.